src/mscorlib/src/System/Threading/WaitHandle.cs

```diff
@@ -448,7 +448,7 @@ public static int WaitAny(WaitHandle[] waitHandles, int millisecondsTimeout)
                                         WaitHandle toWaitOn)
         {
 #if PLATFORM_UNIX
-            throw new PlatformNotSupportedException();
+            throw new PlatformNotSupportedException(SR.Arg_PlatformNotSupported); // https://github.com/dotnet/coreclr/issues/10441
```

Comment:
How are we deciding when to throw a generic error message versus a more specific one?